pkg/cluster/cluster.go

```diff
@@ -188,6 +191,18 @@ func (c *Cluster) checkClientAuthUpdate() error {
 			c.cluster.SetClientAuthSecretResourceVersion(result.ResourceVersion)
 			return c.updateConfigSecret()
 		}
+	} else if c.cluster.Spec.Auth.ClientsAuthFile != "" {
+		clientAuthBytes, err := ioutil.ReadFile(c.cluster.Spec.Auth.ClientsAuthFile)
```

Comment:
The nasty thing about this, it it requires the auth file to be attached to the nats-operator.  So you need to be able to modify it's deployment to change a NatsCluster.  Seems like bad separation.

Originally, I wanted to make this adjust the container cmd to just be `gnatsd -c {{ authFile }}`` However, then all the other configuration options that nats-operator handles gets overridden.  gnatsd doesn't seem to support merging multiple config files together like consul, etc.

Reply:
I think maybe what we can do is to instead use the `Include` option to point to the path of the secret that was mounted via the volume mounts merged from the prev PR:　https://github.com/nats-io/nats-operator/blob/master/pkg/conf/natsconf.go#L27
That would generalize this as well into being able to opt-in override other parts of the config as well if needed (new options in the config file would not override others already done by the operator)

Reply:
Here `Include` is being used to include a generated config file from another container, in this case only containing the advertise external IP address: 
https://github.com/nats-io/nats-operator/blob/bbff7978648d4411c157c16bded3df51a9a017cf/pkg/util/kubernetes/kubernetes.go#L413-L432

This would also mean that if wanting to use a custom volume mount to include a config and also have the external ip address, the config file in the custom volume mount need to also have an `include` to the generated file with the ip address (which I think is ok).

Reply:
Then since it is a mounted file, then the reloader sidecar could track the changes to the file instead of the operator by adding the custom volume mount secret config file here: https://github.com/nats-io/nats-operator/blob/master/pkg/reloader/reloader.go#L94-L99

Reply:
Okay, I misunderstood what you're saying, but I think this is better:

1. Add an include section into the `AuthConfig` to allow users to specify a file to include. 
2. Then using the template, and VolumeMounts stuff from the previous patch, we are able to add an:
    a.  a shared empty volume
    a. initContainer that drops a auth config json file into above volume
    b. use the VolumeMounts option to mount that volume into the nats container.
    c.  It should include the auth config file from there.

So this would result in a NatsCluster CRD like this:

```yaml
apiVersion: "nats.io/v1alpha2"
kind: "NatsCluster"
metadata:
  name: my-nats-cluster
  namespace: nats-stuff
spec:
  size: 5
  version: "1.3.0"
  service:
    maxPayload: 20971520
  pod:
    volumeMounts:
      - name: authconfig
        mountPath: /authconfig
  template: 
    serviceAccountName: my-secrets-getting-sa
    initContainers:
      - name: secret-getter
        image: "...."
        args: ["/authconfig/auth.json"]
        volumeMounts:
          - name: authconfig
            mountPath: /authconfig
    volumes:
      - name: authconfig
        emptyDir: {}
  auth:
    clientAuthFile: "/authconfig/auth.json"
```

Reply:
That looks good to me 💯

Reply:
The includes approach under auth section I think will work well, I should mention though that there was a bug in NATS 1.4.1 and earlier with `include` being ignored after being generated by the operator so I had to do this... 🙈 

https://github.com/nats-io/nats-operator/blob/bbff7978648d4411c157c16bded3df51a9a017cf/pkg/util/kubernetes/kubernetes.go#L428-L432

If you are ok with it you could use one of the edge images where this is fixed to avoid this for now, we should be releasing a new image with the fixes some time this month:

```yaml
---
apiVersion: "nats.io/v1alpha2"
kind: "NatsCluster"
metadata:
  name: "nats-custom"
spec:
  size: 3
  serverImage: "synadia/nats-server"
  version: "edge-2.0.0-RC5"

  pod:
    enableConfigReload: true

  natsConfig:
    debug: true
    trace: true
    maxPayload: 5242880
```

Reply:
This should probably be updated to be ReplaceAll then?
```
bytes.Replace(rawConfig, []byte(`"include":`), []byte("include "), 1)
```

Reply:
Re: https://github.com/nats-io/gnatsd/commit/441d865c1cd34b2f0ab9750e7286d593be132a7c

Making include a special map keyword at the parser level might bite you in the future.

Reply:
right, since there could be more than one include then should change to `ReplaceAll` now. Currently `include` is the only keyword, and we haven't released this change yet... so still time to change it. I added it to try to make the format more compatible with JSON.  One issue from this for example is that since now `include` is a keyword from a map, then can have only a single `include` even though the NATS config allows multiple includes.